cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/GitPATAuthStrategy.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.sourcecontrol;
+
+import io.cdap.cdap.api.security.store.SecureStore;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryConfig;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+
+/**
+ * An {@link AuthStrategy} to use with GitHub and Personal Access Tokens.
+ */
+public class GitPATAuthStrategy implements AuthStrategy {
+  private static final String GITHUB_PAT_USERNAME = "oauth2";
+
+  public GitPATAuthStrategy() {
+  }
+
+  @Override
+  public CredentialsProvider getCredentialProvider(SecureStore store,
+                                                   RepositoryConfig config,
+                                                   NamespaceId namespaceId) throws AuthenticationException {
+
+    try {
+      byte[] bytes = store.get(namespaceId.getNamespace(), config.getAuth().getTokenName()).get();
+      String token = new String(bytes);
```

Comment:
Creating `String` from `byte[]` should always provides a `Charset`.

Reply:
Done.